### ISSUE TITLE
[Bug]:  Py310 unittests on MacOS are failing

### ISSUE THREAD

**Issue by AnandInguva** [apache/beam#23734]
### What happened?

Python 3.10 unit tests are failing on MacOS. Could be related to newer version of `grpcio` released on October 18, 2022.
https://github.com/apache/beam/actions/runs/3278125487/jobs/5396167668

### Issue Priority

Priority: 1

### Issue Component

Component: test-failures

**Comment by AnandInguva** [comment]
.add-labels "failing test"

**Comment by AnandInguva** [comment]
Partial fix: https://github.com/apache/beam/pull/23735 is to pin a version to grpcio only for MacOS. I was able to pass this locally but it fails on GH actiions.

**Comment by damccorm** [comment]
@AnandInguva looks like it is now working with your fix on master - https://github.com/apache/beam/actions/runs/3286759714/jobs/5415213138

**Comment by AnandInguva** [comment]
Yes, we can downgrade this to P2. But still, it doesn't explain why the sdist of grpcio is failing on MacOS.

Is there a way to SSH into the GH workers and check why the sdist failing? @damccorm

**Comment by AnandInguva** [comment]
.add-labels P2

**Comment by damccorm** [comment]
Technically, yes - https://github.com/appleboy/ssh-action - but we can't add that action to the main beam repo because its a big security hole. If you want to go that route, I'd recommend doing it on a fork of the repo. You might be better off just scripting commands to run in your fork and checking them in as actions steps though

**Comment by AnandInguva** [comment]
Thanks Danny. I will try to debug that way.

**Comment by AnandInguva** [comment]
This has been solved in #24599.
